server/datastore/mysql/scheduled_queries.go

```diff
@@ -166,3 +166,15 @@ func (d *Datastore) ScheduledQuery(id uint) (*fleet.ScheduledQuery, error) {
 
 	return sq, nil
 }
+
+func (ds *Datastore) CleanupOrphanScheduledQueryStats() error {
+	_, err := ds.db.Exec(`DELETE FROM scheduled_query_stats where host_id not in (select id from scheduled_queries where id=scheduled_query_id)`)
```

Comment:
Should `host_id` here be `scheduled_query_id`?
```suggestion
	_, err := ds.db.Exec(`DELETE FROM scheduled_query_stats where scheduled_query_id not in (select id from scheduled_queries where id=scheduled_query_id)`)
```

Separately, I'm not sure I understand the subquery. Is there a reason to prefer this over `DELETE FROM scheduled_query_stats WHERE scheduled_query_id NOT IN (SELECT id FROM scheduled_queries)`?

Reply:
Oh crap. You're right... now why did my test pass then... will check. Thanks!